modules/sdk-coin-doge/src/doge.ts

```diff
@@ -78,4 +80,16 @@ export class Doge extends AbstractUtxoCoin {
   ): Promise<CrossChainRecoverySigned<TNumber> | CrossChainRecoveryUnsigned<TNumber>> {
     return super.recoverFromWrongChain(params);
   }
+
+  parseUnspents<TNumber extends number | bigint>(unspents: Unspent<TNumber>[]): Unspent<TNumber>[] {
```

Comment:
The fact that you are casting `(unspent as any)` later is a hint that our types are incorrect. 

```suggestion

// should live in abstractUtxoCoin.ts
/** Unspent data as returned by Wallet Platform */
type UnspentJSON = Unspent & { value: number; valueString: string };

  parseUnspents(unspents: UnspentJSON[]): Unspent<bigint>[] {
```

Reply:
I have to return Unspent<number | bigint> from this method as it will be used by other coins too. And when I do `  parseUnspents<TNumber extends number | bigint>(unspents: UnspentJSON[]): Unspent<TNumber>[] {
    return unspents;
  }`